tools/migration/internal/repo_fs_wrangler_test.go

```diff
@@ -16,71 +17,110 @@ import (
 	tf.UnitTest(t)
 
-	assert := ast.New(t)
-	require := req.New(t)
-
 	t.Run("Uses the option values when passed to ctor", func(t *testing.T) {
-		dirname := "/tmp/filecoindir"
-		mustMakeTmpDir(require, dirname)
-		defer mustRmDir(require, dirname)
+		oldRepo := mustMakeTmpDir(t, "")
+		defer mustRmDir(t, oldRepo)
 
-		rmh := NewRepoFSWrangler(dirname, "", "1", "2")
+		rmh := NewRepoFSWrangler(oldRepo, "", "1", "2")
 		or, err := rmh.GetOldRepo()
-		require.NoError(err)
+		require.NoError(t, err)
 
-		assert.Equal(dirname, or.Name())
+		assert.Equal(t, oldRepo, or.Name())
 	})
 }
 
 func TestRepoMigrationHelper_MakeNewRepo(t *testing.T) {
 	tf.UnitTest(t)
 
-	assert := ast.New(t)
-	require := req.New(t)
+	t.Run("Creates the dir with the right permissions", func(t *testing.T) {
+		oldRepo := mustMakeTmpDir(t, "")
+		defer mustRmDir(t, oldRepo)
 
-	t.Run("Creates the dir", func(t *testing.T) {
-		dirname := "/tmp/myfilecoindir"
-		mustMakeTmpDir(require, dirname)
-		defer mustRmDir(require, dirname)
+		rmh := NewRepoFSWrangler(oldRepo, "", "1", "2")
+		require.NoError(t, rmh.MakeNewRepo())
+		defer mustRmDir(t, rmh.GetNewRepoPath())
 
-		rmh := NewRepoFSWrangler(dirname, "", "1", "2")
-		or, err := rmh.MakeNewRepo()
-		require.NoError(err)
-		defer mustRmDir(require, or.Name())
-		assert.True(fsutils.IsDir(or.Name()))
+		stat, err := os.Stat(rmh.GetNewRepoPath())
+		require.NoError(t, err)
+		expectedPerms := "drwxr--r--"
+		assert.Equal(t, expectedPerms, stat.Mode().String())
 	})
 
 }
 
 func TestGetNewRepoPath(t *testing.T) {
 	tf.UnitTest(t)
-	assert := ast.New(t)
-	require := req.New(t)
 
 	dirname := "/tmp/myfilecoindir"
 
 	t.Run("Uses the new repo opt as a prefix if provided", func(t *testing.T) {
 		rmh := NewRepoFSWrangler(dirname, "/tmp/somethingelse", "1", "2")
 		newpath := rmh.GetNewRepoPath()
 		rgx, err := regexp.Compile("/tmp/somethingelse_1_2_[0-9]{8}-[0-9]{6}$")
-		require.NoError(err)
-		assert.Regexp(rgx, newpath)
+		require.NoError(t, err)
+		assert.Regexp(t, rgx, newpath)
 	})
 
 	t.Run("Adds a timestamp to the new repo dir", func(t *testing.T) {
 		rmh := NewRepoFSWrangler(dirname, "", "1", "2")
 		newpath := rmh.GetNewRepoPath()
 		rgx, err := regexp.Compile("/tmp/myfilecoindir_1_2_[0-9]{8}-[0-9]{6}$")
-		require.NoError(err)
-		assert.Regexp(rgx, newpath)
+		require.NoError(t, err)
+		assert.Regexp(t, rgx, newpath)
 	})
 }
 
-func mustMakeTmpDir(require *req.Assertions, dirname string) {
-	require.NoError(os.Mkdir(dirname, os.ModeDir|os.ModeTemporary|os.ModePerm))
+func TestRepoFSWrangler_MakeNewRepo(t *testing.T) {
+	dirname := mustMakeTmpDir(t, "")
+	rmh := NewRepoFSWrangler(dirname, "", "1", "2")
+	require.NoError(t, rmh.MakeNewRepo())
+	dir, err := os.Open(rmh.GetNewRepoPath())
+	require.NoError(t, err)
+	stat, err := dir.Stat()
+	require.NoError(t, err)
+	expectedPerms := "drwxr--r--"
+	assert.Equal(t, expectedPerms, stat.Mode().String())
+}
+
+func TestRepoFSWrangler_InstallNewRepo(t *testing.T) {
```

Comment:
Mainly this test is to make sure I correctly used go file system calls, but also that permissions were set correctly.